server/geoip_service.py

```diff
@@ -69,23 +70,45 @@ def __init__(self):
             Download the geoip database to a file. If the downloaded file is not
         a valid gzip file, then it does NOT overwrite the old file.
         """
+        if not config.GEO_IP_LICENSE_KEY:
+            self._logger.warning(
+                "GEO_IP_LICENSE_KEY not set! Unable to download GeoIP database!"
+            )
+            return
 
         self._logger.info("Downloading new geoip database")
 
         # Download new file to a temp location
-        temp_file_path = "/tmp/geoip.mmdb.gz"
-        await self._download_file(config.GEO_IP_DATABASE_URL, temp_file_path)
+        temp_file_path = "/tmp/geoip.mmdb.tar.gz"
+        await self._download_file(
+            config.GEO_IP_DATABASE_URL,
+            config.GEO_IP_LICENSE_KEY,
+            temp_file_path
+        )
+
+        def get_db_member(tar: tarfile.TarFile) -> str:
```

Comment:
is there a reason for inner function? Like i mean, you don't use local variables from outer scope, etc...

Reply:
Yes, it's for readability. If I put it inline it looks like this:
```python
with tarfile.open(temp_file_path, 'r:gz') as tar:
    for name in tar.getnames():
        if name.endswith("GeoLite2-Country.mmdb"):
            with open(self.file_path, 'wb') as f_out:
                f_in = tar.extractfile(name)
```
Which is too much indentation IMO

Reply:
We could also do `name = next((n for n in tar.getnames() if n.endswith(foo)), None)`, then throw if it's None.

Reply:
Oh yea, that's pretty clean. There's also a function for getting `TarInfo` objects instead of just names which we can use to make sure that we only look at files and not directories as well.